build/Targets/GenerateCompilerExecutableBindingRedirects.targets

```diff
@@ -13,58 +13,62 @@
-    <SuggestedBindingRedirects Include="System.Console, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Diagnostics.FileVersionInfo, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Diagnostics.StackTrace, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.3.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.IO.Compression, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089">
-      <MaxVersion>4.1.2.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.IO.FileSystem, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.2.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.IO.FileSystem.Primitives, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.2.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.IO.Pipes, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Reflection.Metadata, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>1.4.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Threading.Thread, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Security.Cryptography.Primitives, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Security.Principal.Windows, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.ValueTuple, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51">
-      <MaxVersion>4.0.1.0</MaxVersion>
-    </SuggestedBindingRedirects>
-    <SuggestedBindingRedirects Include="System.Xml.ReaderWriter, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <MaxVersion>4.1.0.0</MaxVersion>
-    </SuggestedBindingRedirects>
-  </ItemGroup>
+  <Target Name="_GenerateCompilerExecutableBindingRedirects" BeforeTargets="GenerateBindingRedirects" DependsOnTargets="GetAssemblyVersion">
+    <Error Text="AssemblyVersion is not initialized" Condition="'$(AssemblyVersion)' == ''"/>
```

Comment:
Why does this fix this? AssemblyVersion is now computed during execution instead of evaluation, and that forces this to be done later too?

Reply:
Correct. It's always been that way in .NET SDK.https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets#L160

Reply:
Is there a reason we still have to manually define these? Why isn't it automatic?

Reply:
I don't know why we have these.